code/modules/antagonists/traitor/equipment/Malf_Modules.dm

```diff
@@ -404,7 +404,12 @@ GLOBAL_LIST_INIT(blacklisted_malf_machines, typecacheof(list(
 
 /datum/AI_Module/large/upgrade_turrets/upgrade(mob/living/silicon/ai/AI)
 	for(var/obj/machinery/porta_turret/ai/turret in GLOB.machines)
-		turret.obj_integrity += 30
+		turret.max_integrity = 200
+		turret.obj_integrity = 200
+		turret.emp_proofing = TRUE
+		turret._AddElement(EMP_PROTECT_SELF | EMP_PROTECT_WIRES, EMP_PROTECT_CONTENTS)
```

Comment:
```suggestion
		turret.AddComponent(/datum/component/empprotection, EMP_PROTECT_SELF | EMP_PROTECT_CONTENTS | EMP_PROTECT_WIRES)
```
This is a component on bee, not an element. Also, if you have this as a component you don't need to override porta_turret/emp_act, since it already has handling for EMP_PROTECT_SELF.